src/Digdir.Domain.Dialogporten.Infrastructure/UnitOfWork.cs

```diff
@@ -57,4 +73,52 @@ public async Task<SaveChangesResult> SaveChangesAsync(CancellationToken cancella
 
         return new Success();
     }
+
+    // Optimistic concurrency
+    // Total timeout for optimistic concurrency handling
+    private const int TimeoutInSeconds = 10;
+    private static readonly AsyncTimeoutPolicy TimeoutPolicy =
+        Policy.TimeoutAsync(TimeoutInSeconds,
+        TimeoutStrategy.Pessimistic,
+        (_, _, _) => throw new OptimisticConcurrencyTimeoutException());
+
+    // Backoff strategy with jitter for retry policy, starting at ~5ms
+    private const int MedianFirstDelayInMs = 5;
+    private static readonly IEnumerable<TimeSpan> JitterDelay = Backoff.DecorrelatedJitterBackoffV2(
+        medianFirstRetryDelay: TimeSpan.FromMilliseconds(MedianFirstDelayInMs),
+        retryCount: int.MaxValue);
+
+    // Fetch the db revision and retry
+    // https://learn.microsoft.com/en-us/ef/core/saving/concurrency?tabs=data-annotations#resolving-concurrency-conflicts
+    private static readonly AsyncRetryPolicy RetryPolicy = Policy
+        .Handle<DbUpdateConcurrencyException>()
+        .WaitAndRetryAsync(JitterDelay, (exception, _) =>
+        {
+            if (exception is not DbUpdateConcurrencyException concurrencyException)
+            {
+                return;
+            }
+
+            foreach (var entry in concurrencyException.Entries)
+            {
+                var dbValues = entry.GetDatabaseValues();
```

Comment:
Du trenger ikke hente det for andre enn entitetene som implementerer IVersionableEntity